File: java-martian-ddd/src/Parsing.java
final class Parsing {
    static int[] grid(String line) {
        String[] p = line.trim().split("\\s+");
        return new int[] { Integer.parseInt(p[0]), Integer.parseInt(p[1]) };
    }

    static Object[] robot(String line) {
        String[] p = line.trim().split("\\s+");
        return new Object[] { new Position(Integer.parseInt(p[0]), Integer.parseInt(p[1])), Orientation.valueOf(p[2]) };
    }

    static String norm(String line) {
        return line.trim().replaceAll("[^LRF]", "");
    }
}

File: langs/ocaml-ddd/parsing.ml
open Position
open Orientation

let ints line = line |> String.trim |> String.split_on_char ' ' |> List.filter (fun s -> s<>"") |> List.map int_of_string

let parse_grid line = match ints line with | [x;y] -> (x,y) | _ -> failwith "bad grid"

let parse_robot line =
  match String.split_on_char ' ' (String.trim line) |> List.filter ((<>) "") with
  | [sx; sy; so] -> ({x=int_of_string sx; y=int_of_string sy}, of_string so)
  | _ -> failwith "bad robot"

let normalize s = String.to_seq s |> Seq.filter (fun c -> c='L' || c='R' || c='F') |> String.of_seq
